### ISSUE TITLE
How do you develop the javascript?

### ISSUE THREAD

**Issue by mauritsvanrees** [collective/collective.taxonomy#147]
Which node / npm / yarn version should I use?  Would be good to have this documented, or configured in the [Volta](https://volta.sh) way.

I ran `yarn install` (version 3.5.0) and got a modified `yarn.lock` and untracked files `.yarn/` and `.yarnrc.yml`.

**Comment by mauritsvanrees** [comment]
With `yarn` version 1 it seems okay, I get no problems, and no changes after running `yarn install && yarn build`.  Node 16 and 18 seem fine.  This Volta config in `package.json` would help then:

```
  "volta": {
    "node": "18.16.0",
    "yarn": "1.22.19"
   }
```

I do see that linting fails:

```
$ yarn lint
yarn run v1.22.19
$ eslint src
Error: Cannot find module 'escope'
Require stack:
- /Users/maurits/clients/vmm/formulieren-plone6/checkouts/collective.taxonomy/src/collective/taxonomy/javascripts/node_modules/eslint/lib/api.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
    at monkeypatch (/Users/maurits/clients/vmm/formulieren-plone6/checkouts/collective.taxonomy/src/collective/taxonomy/javascripts/node_modules/babel-eslint/index.js:41:26)
    at exports.parse (/Users/maurits/clients/vmm/formulieren-plone6/checkouts/collective.taxonomy/src/collective/taxonomy/javascripts/node_modules/babel-eslint/index.js:384:5)
    at parse (/Users/maurits/clients/vmm/formulieren-plone6/checkouts/collective.taxonomy/src/collective/taxonomy/javascripts/node_modules/eslint/lib/linter/linter.js:641:29)
    at Linter._verifyWithoutProcessors (/Users/maurits/clients/vmm/formulieren-plone6/checkouts/collective.taxonomy/src/collective/taxonomy/javascripts/node_modules/eslint/lib/linter/linter.js:1111:33)
    at Linter._verifyWithConfigArray (/Users/maurits/clients/vmm/formulieren-plone6/checkouts/collective.taxonomy/src/collective/taxonomy/javascripts/node_modules/eslint/lib/linter/linter.js:1255:21)
    at Linter.verify (/Users/maurits/clients/vmm/formulieren-plone6/checkouts/collective.taxonomy/src/collective/taxonomy/javascripts/node_modules/eslint/lib/linter/linter.js:1210:25)
    at Linter.verifyAndFix (/Users/maurits/clients/vmm/formulieren-plone6/checkouts/collective.taxonomy/src/collective/taxonomy/javascripts/node_modules/eslint/lib/linter/linter.js:1400:29)
    at verifyText (/Users/maurits/clients/vmm/formulieren-plone6/checkouts/collective.taxonomy/src/collective/taxonomy/javascripts/node_modules/eslint/lib/cli-engine/cli-engine.js:230:48)
    at CLIEngine.executeOnFiles (/Users/maurits/clients/vmm/formulieren-plone6/checkouts/collective.taxonomy/src/collective/taxonomy/javascripts/node_modules/eslint/lib/cli-engine/cli-engine.js:798:28)
error Command failed with exit code 1.
```

But `yarn start` works.

And if I do `export NODE_ENV=development` and then `make run` in the top directory, then the "Edit taxonomy data" page uses the javascript from the live yarn server, so that works.

Just trying to get my bearings in this package.

**Comment by petschki** [comment]
latest yarn `1.x` is used here (1.22.22)

**Comment by petschki** [comment]
I've updated the development resources and README here to make it a bit simpler #160

**Comment by mauritsvanrees** [comment]
Thanks!  Sorry, no time to review until second half of July.

**Comment by petschki** [comment]
I've merged #160 ... feel free to open a new issue if there are other problems regarding development.

**Comment by mauritsvanrees** [comment]
Thanks!
